spec/lib/cyclonedx/report_rust_crates_cyclonedx_spec.rb

```diff
@@ -0,0 +1,213 @@
+      repo = Salus::Repo.new('spec/fixtures/report_rust_crates/lock_only')
+      scanner = Salus::Scanners::ReportRustCrates.new(repository: repo, config: {})
+      scanner.run
+
+      rust_cyclonedx = Cyclonedx::ReportRustCrates.new(scanner.report)
+      expect(rust_cyclonedx.build_components_object[0..2]).to match_array(
+        [
+          {
+            "bom-ref": "pkg:cargo_lock/autocfg@0.1.7",
+            "type": "library",
+            "group": "",
+            "name": "autocfg",
+            "version": "0.1.7",
+            "purl": "pkg:cargo_lock/autocfg@0.1.7",
+            "properties": [
+              {
+                "key": "source",
+                "value": "registry+https://github.com/rust-lang/crates.io-index"
+              },
+              {
+                "key": "dependency_file",
+                "value": "Cargo.lock"
+              }
+            ]
+          },
+          {
+            "bom-ref": "pkg:cargo_lock/bitflags@1.2.1",
+            "type": "library",
+            "group": "",
+            "name": "bitflags",
+            "version": "1.2.1",
+            "purl": "pkg:cargo_lock/bitflags@1.2.1",
+            "properties": [
+              {
+                "key": "source",
+                "value": "registry+https://github.com/rust-lang/crates.io-index"
+              },
+              {
+                "key": "dependency_file",
+                "value": "Cargo.lock"
+              }
+            ]
+          },
+          {
+            "bom-ref": "pkg:cargo_lock/cloudabi@0.0.3",
+            "type": "library",
+            "group": "",
+            "name": "cloudabi",
+            "version": "0.0.3",
+            "purl": "pkg:cargo_lock/cloudabi@0.0.3",
+            "properties": [
+              {
+                "key": "source",
+                "value": "registry+https://github.com/rust-lang/crates.io-index"
+              },
+              {
+                "key": "dependency_file",
+                "value": "Cargo.lock"
+              }
+            ]
+          }
+        ]
+      )
+    end
+
+    it 'should calculate and report all the dependencies in the manifest if Cargo.lock is absent' do
+      repo = Salus::Repo.new('spec/fixtures/report_rust_crates/manifest_only')
+      scanner = Salus::Scanners::ReportRustCrates.new(repository: repo, config: {})
+
+      # We will stub the lock file generate to keep our specs from needing to
+      # hit the internet and pulldown the dependency graph
+      allow(scanner).to receive(:run_shell).with(/cargo tree/) do
+        existing_lock = 'spec/fixtures/report_rust_crates/lock_only/Cargo.lock'
+        mock_lock = File.join(repo.path_to_repo, 'Cargo.lock')
+        FileUtils.cp existing_lock, mock_lock
+      end
+
+      scanner.run
+
+      rust_cyclonedx = Cyclonedx::ReportRustCrates.new(scanner.report)
+      expected = [
+        {
+          "bom-ref": "pkg:cargo_lock/autocfg@0.1.7",
```

Comment:
`"pkg:cargo_lock/autocfg@0.1.7"`

Unsure if "cargo_lock" is the desired type here? Simple to change in report_rust_crates.rb if necessary.

Reply:
Is `cargo_lock` the only `type` for Rust?

Reply:
In report_rust_crates.rb, it seems we're currently extrapolating all dependency info from cargo.lock if it's available or generating it if it's not. 

The type field is hard coded in report_rust_crates.rb to `cargo_lock` similar to other scanners, I've reached out to Maddie about this in general to see what implications it could have w/Dep track

Reply:
Maddie got back to me and let me know that the type in the purl field is sent by Dep track to other programs to get more data. 

I'll change this to match list of defined types @ https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst

and do likewise for other scanners 👍